### PR TITLE
[IO] Support multi dimenional label for record io

### PR DESCRIPTION
**Pull request by piiswrong**
@winstywang Also reverted previous PR because it changes behavior. Added windows support

### CODE REVIEW

**Comment by winstywang** [comment]
@Trangle Could you help review this PR and test it in Windows?
For multi label part, I will review it. @piiswrong

**Comment by Trangle** [comment]
@winstywang OK. I have already test it in Windows.

**Comment by piiswrong** [comment]
@winstywang

File: tools/im2rec.py [hunk 1]
```diff
@@ -1,7 +1,4 @@
 # -*- coding: utf-8 -*-
-# add file encoding here
-
-from __future__ import print_function
```
Comment (Trangle) [inline tools/im2rec.py]: For compatible, this may be OK to be reserved

File: tools/im2rec.py [hunk 2]
```diff
@@ -68,34 +66,23 @@ def make_list(args):
...
-    return image_list
-
-
-# Changed the original function write_record cause the multiprocessing must be in the __main__ process in Windows, otherwise the Pickle
-# Error would happen
+            item = [int(line[0])] + [line[-1]] + [float(i) for i in line[1:-1]]
+            yield item
 
 def image_encode(args, item, q_out):
-    # move the coler modes here
```
Comment (Trangle) [inline tools/im2rec.py]: this is transform args.color to cv2 macros and reserved it ?

**Comment by Trangle** [comment]
@winstywang @piiswrong , Sorry that i can't run it(im2rec.py) through like this command:
python im2rec.py test ../example/test_images/data/ --saving-folder=../example/test_images/data80 --resize=80 --num-thread=12

I put my images in ../example/test_images/data/test

and the log may like this:
`['command_py.txt', 'im2rec.cc', 'im2rec.py', 'im2rec.py.bak', 'im2rec.pyc', 'kill-mxnet.py', 'launch.py', 'parse_log.py', 'test.lst']
('Creating .rec file from', 'test.lst', 'in', '../example/test_images/data80')
('read none error:', 'data/test/1.jpg')
('read none error:', 'data/test/2.jpg')
('read none error:', 'data/test/3.jpg')
('read none error:', 'data/test/4.jpg')
('read none error:', 'data/test/5.jpg')
('read n(o'nree aedr rnoorn:e' e, r'rdoart:a'/, t'edsatta//5t4e9s0t7/.1jp3g8(6''0r)3e
.ajdp( g''rn)o
enaed ( e'nrroenaredo r e:nr'ro, no'er :de'r, rao'trda:a/'tt, a'/etdesastt/ta7/4/9961t52e1.s0tj./j1pp5gg7'6)3
')9.(
j'prg(e'a'read)d
 n(no'norene eae der rrnrooornr::e'' , e, 'r'drdoaartt:a'a/, /t'tedesastt/ta1//01t74e00s6t82/6.92j4.9p6jg4'p).g
j'p)g
('')(r
'era(ed'a rdneo anndeo  enrnoreno er e:e'rr, rr'oodrra::t'', a, '/'dtdaetasatt//at1/1te3es7st4t.//j46p69g954'453)..
jj(pp'ggr'')e()`

File: tools/im2rec.py [hunk 3]
```diff
@@ -110,48 +97,42 @@ def image_encode(args, item, q_out):
...
 def read_worker(args, q_in, q_out):
     while not q_in.empty():
         item = q_in.get()
         image_encode(args, item, q_out)
 
-# the write_worker 
 def write_worker(q_out, fname, saving_folder):
     pre_time = time.time()
     sink = []
+    os.chdir(saving_folder)
```
Comment (Trangle) [inline tools/im2rec.py]: folder changed may cause to files not found as i meet before

File: tools/im2rec.py [hunk 4]
```diff
@@ -68,34 +66,23 @@ def make_list(args):
...
+            item = [int(line[0])] + [line[-1]] + [float(i) for i in line[1:-1]]
+            yield item
 
 def image_encode(args, item, q_out):
-    # move the coler modes here
-    color_modes = {-1: cv2.IMREAD_UNCHANGED,
-                   0: cv2.IMREAD_GRAYSCALE,
-                   1: cv2.IMREAD_COLOR}
-
-    # cause the content of make_list, here get the parient directory of data root
```
Comment (Trangle) [inline tools/im2rec.py]: this may be relevant to the generate '.lst' process

File: tools/im2rec.py (hunk 3 above)
Comment (Trangle) [inline tools/im2rec.py]: the same reason like above (folder change)

**Comment by piiswrong** [comment]
try using absolute path for root

**Comment by Trangle** [comment]
@piiswrong I tried absolute path, and prefix is not recognized.
`python im2rec.py va D:/MyCoding/DeepLearning/test_example/mxnet/example/kaggle-ndsb1/data/ --saving-folder=../example/kaggle-ndsb1/data48 --resize=48 --num-thread=12
['command_py.txt', 'im2rec.cc', 'im2rec.py', 'im2rec.py.old', 'im2rec.pyc', 'kill-mxnet.py', 'launch.py', 'parse_log.py', 'test.lst']
Did not find and list file with prefix va
`

File: tools/im2rec.py [hunk 5]
```diff
@@ -199,68 +180,58 @@ def main():
...
-            _, s, _ = q_out.get()
-            record.write(s)
-            cnt += 1
-            if cnt % 1000 == 0:
-                cur_time = time.time()
-                print ('time:', cur_time - pre_time, ' count:', cnt)
-                pre_time = cur_time
-    # add total print operation
-    print ('total: ', len(source))
+        files = [fname for fname in os.listdir('.') if os.path.isfile(fname)]
```
Comment (Trangle) [inline tools/im2rec.py]: files = [f for f in os.listdir(args.root) if os.path.isfile(os.path.join(args.root, f))]

would be work

**Pull request merged**